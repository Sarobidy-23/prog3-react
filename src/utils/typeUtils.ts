export type todoElement = {
    name: string,
    isDone: boolean
  }

export type itemPropType = {
    element: todoElement,
    onCheck: (index: number)=>void,
    index: number
}