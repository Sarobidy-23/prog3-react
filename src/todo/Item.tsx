import React from 'react';
import "./item.css";
import { itemPropType } from '../utils/typeUtils';

function Item(props: itemPropType) {
    const {element, onCheck, index} = props

    return (
        <div className='item-container'>
            {element.name}
            {!element.isDone && <input type="checkbox" onChange={()=>onCheck(index)}/>}
        </div>
    )
}

export default Item
