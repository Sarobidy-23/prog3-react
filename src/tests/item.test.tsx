import renderer, { ReactTestRendererJSON, ReactTestRendererNode } from 'react-test-renderer';
import Item from '../todo/Item';
import { todoElement } from '../utils/typeUtils';

const element: todoElement ={"name":"todoItem","isDone":false};
const check = ()=>{};

it("Snapshot: render Item component", ()=>{
    const tree = renderer.create(
        <Item onCheck={check} index={1} element={element}/>)
    .toJSON();
    expect(tree).toMatchSnapshot();
})

it("Item text is present in screen", ()=>{
    const tree = renderer.create(
        <Item onCheck={jest.fn()} index={1} element={element}/>
    ).toJSON() as ReactTestRendererJSON;
    
    expect(tree.children![0]).toBe("todoItem")
})