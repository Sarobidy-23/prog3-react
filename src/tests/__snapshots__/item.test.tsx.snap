// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot: render Item component 1`] = `
<div
  className="item-container"
>
  todoItem
  <input
    onChange={[Function]}
    type="checkbox"
  />
</div>
`;
